projects/packages/jetpack-mu-wpcom/src/features/wpcom-wpadmin-page-view/wpcom-wpadmin-page-view.php

```diff
@@ -0,0 +1,188 @@
+			user_type: '<?php echo esc_js( implode( ',', $user_types ) ); ?>'
+		};
+		_tkq = window._tkq || [];
+		_tkq.push( [ 'identifyUser', <?php echo (int) $current_user->ID; ?>, '<?php echo esc_js( $current_user->user_login ); ?>' ] );
+		_tkq.push( [ 'recordEvent', 'wpcom_admin_page_view', _admin_pv_props ] );
+	</script>
+	<?php
+}
+add_action( 'admin_footer', __NAMESPACE__ . '\wpcom_nosara_track_admin_page_views' );
+
+/**
+ * Track non-calypso Customizer views if the user is linked from the frontend.
+ */
+function wpcom_maybe_track_customizer() {
+	// not when the Customizer is loaded in Calypso
+	if ( isset( $_GET['calypso'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		return;
+	}
+
+	// links from frontend
+	if ( isset( $_GET['url'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		return wpcom_track_customizer_from_frontend();
+	}
+
+	// standard wp-admin
+	wpcom_nosara_track_admin_page_views();
+}
+
+/**
+ * Since links to the Customizer from the frontend aren't properly wp-admin (no wp-admin chrome,
+ * you return to the frontend after closing), they shouldn't be counted as wp-admin views.
+ */
+function wpcom_track_customizer_from_frontend() {
+	global $current_user;
+
+	// Since we view so many screens to offer support and do feature dev, I think it's best not to count us.
+	if ( do_not_track_a11ns() ) {
+		return;
+	}
+	?>
+	<script type="text/javascript">
+		_tkq = window._tkq || [];
+		_tkq.push( [ 'identifyUser', <?php echo (int) $current_user->ID; ?>, '<?php echo esc_js( $current_user->user_login ); ?>' ] );
+		_tkq.push( [ 'recordEvent', 'wpcom_customize_loaded_from_frontend' ] );
+	</script>
+	<?php
+}
+add_action( 'customize_controls_print_footer_scripts', __NAMESPACE__ . '\wpcom_maybe_track_customizer' );
+
+/**
+ * Determine if the current user should not be tracked.
+ *
+ * @return bool
+ */
+function do_not_track_a11ns() {
+	$is_simple_site = defined( 'IS_WPCOM' ) && IS_WPCOM;
+	$is_atomic_site = ! $is_simple_site;
+
+	if ( $is_simple_site ) {
+		// Since we view so many screens to offer support and do feature dev, I think it's best not to count us.
+		if ( is_automattician() ) {
+			return true;
+		}
+
+		// Shouldn't hit this because of the is_automattician() check above, but let's be cautious.
+		if ( is_network_admin() ) {
+			return true;
+		}
+	}
+
+	if ( $is_atomic_site ) {
+		return wpcom_atomic_maybe_is_a11n();
+	}
+}
+
+/**
+ * Check if the user might be an a11n on Atomic sites.
+ *
+ * @return bool
+ */
+function wpcom_atomic_maybe_is_a11n() {
+	$is_proxy_atomic    = defined( 'AT_PROXIED_REQUEST' ) && AT_PROXIED_REQUEST;
+	$is_support_session = WPCOMSH_Support_Session_Detect::is_probably_support_session();
+
+	return $is_proxy_atomic && ! $is_support_session;
+}
+
+/**
+ * Retrieves the user types for Atomic sites.
+ *
+ * @return array An array of user types.
+ */
+function wpcom_atomic_get_user_types() {
```

Comment:
wpcom code for the `user_types`: https://github.a8c.com/Automattic/wpcom/blob/dfe6a5d01904c4e447dd922cc045cda1ad53e1b3/wp-content/lib/wpcom-user-type.php#L55-L78